mindsdb/interfaces/skills/skill_tool.py

```diff
@@ -66,7 +73,22 @@ def _make_text_to_sql_tools(self, skill: db.Skills) -> dict:
         return dict(
             name='sql_db_query',
             func=sql_agent.query_safe,
-            description=description
+            description=description,
+            type=skill.type
+        )
+
+    def _make_retrieval_tools(self, skill: db.Skills) -> dict:
+        """
+        creates advanced retrieval tool i.e. RAG
+        """
+        params = skill.params
+        return dict(
+            name=params.get('name', skill.name),
+            config=params.get('retriever_config', {}),
+            description=f'You must use this tool to get more context or information '
+                        f'to answer a question about {params["description"]}. '
+                        f'The input should be the exact question the user is asking.',
```

Comment:
updated with base template for now, we can replace with llm gen if needed but this suffices for now